staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -252,6 +227,56 @@ func (s *Serializer) doEncode(obj runtime.Object, w io.Writer) error {
 	return encoder.Encode(obj)
 }
 
+func (s *Serializer) unmarshal(into runtime.Object, data, originalData []byte, notRegistered bool) error {
+	// If the deserializer is non-strict, return here.
+	if !s.options.Strict {
+		if err := kjson.UnmarshalCaseSensitivePreserveInts(data, into); err != nil {
+			return err
+		}
+		return nil
+	}
+	if notRegistered {
+		strictErrs, err := kjson.UnmarshalStrict(data, into)
+		if err != nil {
+			return err
+		} else if len(strictErrs) > 0 {
+			return runtime.NewStrictDecodingError(strictErrs)
+		}
+		return nil
+	}
```

Comment:
I'm trying to remember why this needs a distinct path... why wouldn't we use the logic below?

Reply:
I don't remember either. Fixed